pkg/skaffold/kubernetes/portforward/entry_manager_test.go

```diff
@@ -62,6 +64,10 @@ func TestStop(t *testing.T) {
 	em.forwardedResources.Store("pod-resource-default-0", pfe1)
 	em.forwardedResources.Store("pod-resource2-default-0", pfe2)
 
+	em.forwardedPorts = &sync.Map{}
+	em.forwardedPorts.Store(9000, struct{}{})
```

Comment:
Once a goroutine has selected an unused port, it will store it in this map. Before other goroutines try and use that port, they will check and make sure it doesn't exist in the map, so multiple goroutines shouldn't find the same port unused. 

This is also convenient because it allows us to use the [GetAvailablePort](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/util/port.go#L37) function which takes in a sync.Map and is also used by the event API, so I'd prefer not to change it, unless you feel strongly. WDYT?

Reply:
like you mentioned. "Once a goroutine has selected an unused port, it will store it in this map" The bug is when multiple routines happen to detect it as unused at the same time.  Does "GetAvailablePort" get called in go subroutines?
happy to chat offline.

Reply:
We should probably introduce our own types that wrap a `map` and a lock. That would make the rest of the code easier to read and maintain.

Reply:
Makes sense, thanks @tejal29  and @dgageot for the explanations! I've updated this.